modules/KoreBuild.Tasks/Utilities/DownloadFileHelper.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Utilities;
+using Task = System.Threading.Tasks.Task;
+
+namespace KoreBuild.Tasks.Utilities
+{
+    public class DownloadFileHelper
+    {
+        public static async Task<bool> DownloadFileAsync(string uri, string destinationPath, bool overwrite, CancellationTokenSource cts, int timeoutSeconds, TaskLoggingHelper log)
```

Comment:
Nit change: `CancellationTokenSource` to `CancellationToken` and pass in `cts.Token`.